### PR TITLE
Properly cache http profiler

### PR DESCRIPTION
**Pull request by ofek**
### What does this PR do?

Resolves https://github.com/DataDog/integrations-extras/issues/216

### Motivation

The `FilebeatCheckHttpProfiler` is supposed to collect and compute deltas between runs but since we were creating a new instance every check run it was unable to do so, thus not reporting some metrics.

### CODE REVIEW

File: filebeat/check.py [hunk 1]
```diff
@@ -205,11 +206,19 @@ class FilebeatCheck(AgentCheck):
 
     def __init__(self, *args, **kwargs):
         AgentCheck.__init__(self, *args, **kwargs)
+        self.instance_cache = {}
 
     def check(self, instance):
-        config = FilebeatCheckInstanceConfig(instance)
+        instance_key = hash_mutable(instance)
```
[the remark this answers is not in the document]
Comment (ofek) [reply filebeat/check.py]: What's the point and how is that safer?

[the remark this answers is not in the document]
Comment (nmuesch) [reply filebeat/check.py]: We communicated about this offline, but my original thought was that it was safer to use the `name` param as its possible the `instance` could be modified at some point (changing the hash). However the hash is taken at the very beginning of the check so this shouldn't really be a concern.

**Comment by danielkasen** [comment]
Just tried this on 6.5.1 and i'm getting the gauge metrics now but none of the INCREMENT_METRIC_NAMES are showing up

**Comment by ofek** [comment]
I've re-opened #216

**Pull request merged**